api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -147,4 +148,23 @@ message LocalRateLimit {
   // of the default ``UNAVAILABLE`` gRPC code for a rate limited gRPC call. The
   // HTTP code will be 200 for a gRPC response.
   bool rate_limited_as_resource_exhausted = 15;
+
+  // Rate limit configuration that used to generate a list of description entries based on
+  // the request context. And the generated entries will be used to find one or multiple matched rate
+  // limit rule from the ``descriptors``.
+  // If this is set, then
+  // :ref:`VirtualHost.rate_limits<envoy_v3_api_field_config.route.v3.VirtualHost.rate_limits>` or
+  // :ref:`RouteAction.rate_limits<envoy_v3_api_field_config.route.v3.RouteAction.rate_limits>` fields
+  // will be ignored.
+  //
+  // .. note::
+  //   Not all configuration fields of
+  //   :ref:`rate limit config <envoy_v3_api_msg_config.route.v3.RateLimit>` is supported at here.
+  //   Following fields are not supported:
+  //
+  //   1. :ref:`rate limit stage <envoy_v3_api_field_config.route.v3.RateLimit.stage>`.
+  //   2. :ref:`dynamic metadata <envoy_v3_api_field_config.route.v3.RateLimit.Action.dynamic_metadata>`.
+  //   3. :ref:`disable_key <envoy_v3_api_field_config.route.v3.RateLimit.disable_key>`.
+  //   4. :ref:`override limit <envoy_v3_api_field_config.route.v3.RateLimit.limit>`.
```

Comment:
I guess some of them are not supported because they are not implemented yet, and not because there's some design blocker.
Can you separate the 2 categories, and make that explicit? This will allow future upgrades (PRs) that enable these features.

Reply:
Hmmm, I think all these will be ignored forever. This is why l initially copied whole RateLimit to a new message and removed them in the new message.

Because historical reason, at the initial design, the ratelimit configuration is split to two parts: filter configuration and `rate_limits` in the route. We we may have multiple ratelimit filters in the filter chain, but there is only one `rate_limits` field in single route.
To ensure multiple limit filter could share same `rate_limits` correctly, the `stage` is introduced. Different filters may have different `stage`. Only the matched policies in the policy list will be applied.

The `stage` is unnecessary because the new `rate_limits` is part of filter configuration. We needn't the `stage` match now.

The `Limit` is only used by the remote ratelimit filter, it make no sense for local rate limit.

The `disable_key` allow us to disable single policy. But the local ratelimit has provided filter level runtime disable key. I have no plan to support this more fine-grained control because it is hard to say it actually make sense.
IMO, we abused the runtime switch. It is pretty good to guard the behavior changes.

But I think it should not be used as general switch of feature or configuration. XDS is good enough for most cases to update configuration dynamically.

Reply:
I think it would be good to reject/warn when there's config that has those fields

Reply:
> I think it would be good to reject/warn when there's config that has those fields

I will add a warn log at the implementation.